### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Vittorio-Caggiano**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Quick update of the methods currently available in the 0.3 release

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,14 +10,21 @@ FairScale is a PyTorch extension library for high performance and large scale tr
...
+   * _experimental_ AmpNet (fairscale.experimental.nn.ampnet_pipe)
 * Sharded training:
-   * Optimizer state sharding (fairscale.optim.oss)
-   * Sharded grad scaler - automatic mixed precision
-   * Sharded distributed data parallel
+   * Optimizer state sharding (fairscale.optim.OSS)
+   * Sharded grad scaler - automatic mixed precision (fairscale.optim.grad_scaler)
+   * Sharded distributed data parallel (fairscale.nn.ShardedDataParallel)
+   * Fully Sharded Data Parallel (FSDP) (fairscale.nn.FullyShardedDataParallel)
+   * _experimental_ Offload (fairscale.experimental.nn.offload)
```
Comment (anj-s) [inline README.md:23]: One thing to note is that this is not part of the pip package yet. Should we hold off mentioning this until the next release? I was going to add release notes and an usage example.

File: README.md [hunk 2]
```diff
@@ -184,3 +191,5 @@ Here is a list of all authors on relevant research papers this work is based on:
 * ZeRO: Samyam Rajbhandari, Jeff Rasley, Olatunji Ruwase, Yuxiong He. [[Paper](https://arxiv.org/pdf/1910.02054.pdf)] [[Code](https://github.com/microsoft/DeepSpeed)]
 * Megatron-LM: Mohammad Shoeybi, Mostofa Patwary, Raul Puri, Patrick LeGresley, Jared Casper, Bryan Catanzaro. [[Paper](https://arxiv.org/pdf/1909.08053.pdf)][[Code](https://github.com/NVIDIA/Megatron-LM)]
 * AdaScale SGD: Tyler B. Johnson, Pulkit Agrawal, Haijie Gu, Carlos Guestrin. [[Paper](https://proceedings.icml.cc/static/paper_files/icml/2020/4682-Paper.pdf)]
+* GShard: Dmitry Lepikhin, HyoukJoong Lee, Yuanzhong Xu, Dehao Chen, Orhan Firat, Yanping Huang, Maxim Krikun, Noam Shazeer, Zhifeng Chen [[Paper]](https://arxiv.org/abs/2006.16668)
+* AMPNet:Alexander L. Gaunt, Matthew A. Johnson, Maik Riechert, Daniel Tarlow, Ryota Tomioka, Dimitrios Vytiniotis, Sam Webster [[Paper]](https://arxiv.org/abs/1705.09786)
```
Comment (anj-s) [inline README.md:195]: We should also add the Zero-offload and L2L paper as references here.

File: README.md (hunk 1 above)
Comment (Vittorio-Caggiano) [reply README.md:23]: yes... let's add it with next release

**Comment by anj-s** [comment]
nit: @Vittorio-Caggiano The PR is in draft mode which might confuse users. Also, could update the PR description section? Since it self explanatory, maybe referring to the PR title is sufficient.

**Review by anj-s — APPROVED** [review]

**Pull request merged**